object_goslice_test.go

```diff
@@ -271,3 +271,138 @@ func TestGoSliceToString(t *testing.T) {
 		t.Fatal(exp)
 	}
 }
+
+func TestGoSliceMemUsage(t *testing.T) {
+	vm := New()
+	tests := []struct {
+		name           string
+		val            *objectGoSlice
+		threshold      int
+		expectedMem    uint64
+		expectedNewMem uint64
+		errExpected    error
+	}{
+		{
+			name:      "should account for each value given a non-empty slice",
+			threshold: 100,
+			val: &objectGoSlice{
+				baseObject: baseObject{
+					val: &Object{runtime: vm},
+				},
+				data: &[]interface{}{
+					valueInt(99),
+					valueInt(99),
+					valueInt(99),
+					valueInt(99),
+				},
+			},
+			// overhead + values
+			expectedMem: SizeEmptyStruct + SizeInt*4,
+			// overhead + values
+			expectedNewMem: SizeEmptyStruct + SizeInt*4,
+			errExpected:    nil,
+		},
```

Comment:
[nit] Similar to the other one I would add a test case exactly like this where values are not wrapped in `valueInt`